File: week1/Arrays/Level2/MaxSubArraySum.java
import java.util.Scanner;

public class MaxSubArraySum {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        for(int i=0;i<n;i++) arr[i] = sc.nextInt();
        int res = Integer.MIN_VALUE;
        int sum = 0;
        for(int num:arr){
            sum = Math.max(sum+num,num);
            res = Math.max(res,sum);
        }
        System.out.println(res);
    }
}

File: week1/Arrays/Level2/MaxSubArraySum.js
const readline = require('readline');
const maxSubArraySum = (nums) => {
    let sum = 0, res = -Infinity;
    for(const num of nums){
        sum = Math.max(sum+num,num);
        res = Math.max(res,sum);
    }
    return res;
};

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("",(userInput) => {
    const nums = userInput.split(" ").map(Number);
    console.log(maxSubArraySum(nums));
    rl.close();
})